app/controllers/api/base_controller/generic.rb

```diff
@@ -28,6 +29,12 @@ def add_resource(type, _id, data)
         end
       end
 
+      def validate_type(klass, type)
+        klass.descendant_get(type)
+      rescue => err
+        raise BadRequestError, "Invalid type #{type} specified - #{err}"
+      end
```

Comment:
This method is a helper and not intended for public consumption, so let's throw it under `private` below.